contracts/pcv/balancer/BalancerPCVDepositBase.sol

```diff
@@ -0,0 +1,168 @@
+    address public immutable poolAddress;
+
+    /// @notice cache of the assets in the Balancer pool
+    IAsset[] internal poolAssets;
+
+    /// @notice the balancer vault
+    IVault public immutable vault;
+
+    /// @notice the balancer rewards contract to claim incentives
+    IMerkleOrchard public immutable rewards;
+
+    /// @notice Balancer PCV Deposit constructor
+    /// @param _core Fei Core for reference
+    /// @param _vault Balancer vault
+    /// @param _rewards Balancer rewards (the MerkleOrchard)
+    /// @param _poolId Balancer poolId to deposit in
+    /// @param _maximumSlippageBasisPoints Maximum slippage basis points when depositing
+    constructor(
+        address _core,
+        address _vault,
+        address _rewards,
+        bytes32 _poolId,
+        uint256 _maximumSlippageBasisPoints
+    ) CoreRef(_core) {
+        vault = IVault(_vault);
+        rewards = IMerkleOrchard(_rewards);
+        maximumSlippageBasisPoints = _maximumSlippageBasisPoints;
+        poolId = _poolId;
+
+        (poolAddress, ) = IVault(_vault).getPool(_poolId);
+
+        // get the balancer pool tokens
+        IERC20[] memory tokens;
+        (tokens, , ) = IVault(_vault).getPoolTokens(_poolId);
+
+        // cache the balancer pool tokens as Assets
+        poolAssets = new IAsset[](tokens.length);
+        for (uint256 i = 0; i < tokens.length; i++) {
+            poolAssets[i] = IAsset(address(tokens[i]));
+        }
+    }
+
+    // Accept ETH transfers
+    receive() external payable {}
+
+    /// @notice Wraps all ETH held by the contract to WETH
+    /// Anyone can call it.
+    /// Balancer uses WETH in its pools, and not ETH.
+    function wrapETH() external {
+        uint256 ethBalance = address(this).balance;
+        if (ethBalance != 0) {
+            Constants.WETH.deposit{value: ethBalance}();
+        }
+    }
+
+    /// @notice unwrap WETH on the contract, for instance before
+    /// sending to another PCVDeposit that needs pure ETH.
+    /// Balancer uses WETH in its pools, and not ETH.
+    function unwrapETH() external onlyPCVController {
+        uint256 wethBalance = IERC20(address(Constants.WETH)).balanceOf(address(this));
+        if (wethBalance != 0) {
+            Constants.WETH.withdraw(wethBalance);
+        }
+    }
+
+    /// @notice Sets the maximum slippage vs 1:1 price accepted during withdraw.
+    /// @param _maximumSlippageBasisPoints the maximum slippage expressed in basis points (1/10_000)
+    function setMaximumSlippage(uint256 _maximumSlippageBasisPoints) external onlyGovernorOrAdmin {
+        require(_maximumSlippageBasisPoints <= Constants.BASIS_POINTS_GRANULARITY, "BalancerPCVDepositBase: Exceeds bp granularity.");
+        maximumSlippageBasisPoints = _maximumSlippageBasisPoints;
+        emit UpdateMaximumSlippage(_maximumSlippageBasisPoints);
+    }
+
+    /// @notice redeeem all assets from LP pool
+    function exitPool() external whenNotPaused onlyPCVController {
```

Comment:
why not include a `to` address?

Reply:
because we want to burn the FEI if the pool contains FEI, and if we exitPool on a target directly, it may not be able to burn FEI held or report it properly as protocol-owned. That was my reasoning

Reply:
Still think we should give ourselves the option